x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java

```diff
@@ -100,7 +101,10 @@ public TransformPersistentTasksExecutor(Client client,
             return new PersistentTasksCustomMetaData.Assignment(null, reason);
         }
         DiscoveryNode discoveryNode = selectLeastLoadedNode(clusterState, (node) ->
-            node.isDataNode() && node.getVersion().onOrAfter(params.getVersion())
+            node.isDataNode() &&
+            // see gh#48019 disable assignment if any node is using 7.2 or 7.3
+            clusterState.getNodes().getMinNodeVersion().onOrAfter(Version.V_7_4_0) &&
```

Comment:
I think this condition should be outside of the `selectLeastLoadedNode` function. 

It is also good to return a new custom assignment explaining why we are not assigning to a node (e.g. `Not assigning persistent task as cluster contains nodes with version older than 7.4.0`).